code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -131,11 +195,74 @@ func (msc *MinerSmartContract) GetMinersStateHandler(ctx context.Context, params
 const cantGetShardersListMsg = "can't get sharders list"
 
 func (msc *MinerSmartContract) GetSharderListHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
-	allShardersList, err := getAllShardersList(balances)
+	var (
+		offsetString = params.Get("offset")
+		limitString  = params.Get("limit")
+		activeString = params.Get("active")
+		offset       = -1
+		limit        = -1
+		err          error
+	)
+	if offsetString != "" {
+		offset, err = strconv.Atoi(offsetString)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if limitString != "" {
+		limit, err = strconv.Atoi(limitString)
+		if err != nil {
+			return nil, err
+		}
+	}
+	filter := event.SharderQuery{}
+	if activeString != "" {
+		active, err := strconv.ParseBool(activeString)
+		if err != nil {
+			return nil, err
+		}
+		filter.Active = null.BoolFrom(active)
+	}
+
+	if balances.GetEventDB() == nil {
+		return nil, common.NewErrInternal("can't get miners. no db connection")
+	}
+
+	sharders, err := balances.GetEventDB().GetShardersWithFilterAndPagination(filter, offset, limit)
 	if err != nil {
-		return "", common.NewErrInternal(cantGetShardersListMsg, err.Error())
+		return "", common.NewErrInternal("can't get miners list", err.Error())
 	}
-	return allShardersList, nil
+	shardersArr := make([]interface{}, len(sharders))
+	for i, sharder := range sharders {
+		shardersArr[i] = map[string]interface{}{
```

Comment:
please don't add new structs
use existing

Reply:
needed for compatibility (old json format)

Reply:
use existing transformer and existing structs
no duplications of struct

Reply:
done